### PR TITLE
feat(reflect): Add method to modify the schema after processing

### PR DESCRIPTION
**Pull request by webdestroya**
This PR is very similar to #52 but moves the call to _after_ the default schema builder has run.

This allows struct properties to be modified easily.

### CODE REVIEW

File: reflect.go [hunk 1]
```diff
@@ -108,7 +108,14 @@ type customSchemaImpl interface {
 	JSONSchema() *Schema
 }
 
+// Function to be run after the schema has been generated.
+// this will let you modify a schema afterwards
+type postSchemaImpl interface {
+	JSONSchemaPost(*Schema)
```
Comment (samlown) [inline reflect.go:114]: I'm not totally convinced on the name...  "Post" just isn't obvious enough in my opinion.

**Review by samlown — COMMENTED** [review]
Looks good! But I'm not convinced the interface method name is obvious enough. "Post" just doesn't say much without more context. How about `JSONSchemaMod` or `JSONSchemaExtend`?

File: reflect.go (hunk 1 above)
Comment (webdestroya) [reply reflect.go:114]: I think `JSONSchemaExtend` is best. If you're good with that I can rename it

Comment (samlown) [reply reflect.go:114]: `JSONSchemaExtension` is perhaps more grammatically and semantically correct :-) (naming things 🤦 )

Comment (webdestroya) [reply reflect.go:114]: I liked `JSONSchemaExtend` better, but your choice. Just let me know your final answer and I'll update the PR

Comment (samlown) [reply reflect.go:114]: Cool, lets go with `JSONSchemaExtend`!

Comment (webdestroya) [reply reflect.go]: Alright, renamed

File: reflect.go [hunk 2]
```diff
@@ -108,7 +108,14 @@ type customSchemaImpl interface {
...
 }
 
+// Function to be run after the schema has been generated.
+// this will let you modify a schema afterwards
+type extendSchemaImpl interface {
+	JSONSchemaExtend(*Schema)
+}
+
 var customType = reflect.TypeOf((*customSchemaImpl)(nil)).Elem()
+var extendType = reflect.TypeOf((*extendSchemaImpl)(nil)).Elem()
```
Comment (samlown) [inline reflect.go:118]: It's the same length as `customType` 😍

**Review by samlown — APPROVED** [review]
Awesome. Thanks!

**Pull request merged**